Start Scene basic implementation

rtherrie opened the pull request:
I added a basic implementation of the start scene. The start scene now matches the wireframe. There are new see rules and see settings buttons that can be implemented to work with new rules and settings scenes. 

addresses issues #21 and #12

bgoh1 reviewed (changes requested):
- Is there a particular reason why the color values & variables are commented out & unused? Would do wonders for clarity if actual intuitive variable names were used, in which there would be no need to look up & copy-paste the hex values everytime

- In fact, I think it would be best if all the colors were declared as constants as in src/game-objects/constants.js then simply import them so that the same colors can be universally across all files. In fact, FAWN & MAHOGANY are already present in constants.js

- Capitalize settings.js into Settings.js. Is Settings an overlay? If so, rename it to SettingsOverlay to keep the naming convention consistent with RulesOverlay. In fact, we might want to make an overall Overlay class which branches off into the different types of overlays (which can be done later)

- Is it expected that the "Settings" & "See Rules" buttons in the Start Screen is dysfunctional for now?

rtherrie commented:
I have addressed your requests

> * Is there a particular reason why the color values & variables are commented out & unused?
I changed it and added the colors as constants. Some scenes worked with Hexadecimal Integer Notation (0x000000) and others worked with Hexadecimal Color Notation (#000000). This might be something fixable later, but for now I made 2 different constants to accommodate each type (EX: FAWN and FAWNHEX)
. 
> * Capitalize settings.js into Settings.js. For that matter, is Settings an overlay? If so, rename it to SettingsOverlay to keep the naming convention consistent with RulesOverlay. In fact, we might want to make an overall Overlay class which branches off into the different types of overlays (which can be done later)
The reasoning behind the names of the files is so that they match the names that the people working on #10 and #3 used. This way merging will be easier when they are done. I believe that changing the names to be more uniform is something that should be done _after_ their code has been merged, as this will make the merging process easier with less conflicts. 

> * Is it expected that the "Settings" & "See Rules" buttons in the Start Screen is dysfunctional for now?
Yes, the "Settings" & "See Rules" buttons are expected to be dysfunctional. They take you to a grey screen because the code for the settings and rules overlays has not been implemented yet, as that work has been assigned in issues #10 and #3, not issues #21 and #12 , which are the issues this pull request addresses

bgoh1 reviewed (approved):
Alright, looks good (though I'm not a fan of the comment format where the comments are one space away from the code; feels like the comments are harder to read)
